### ISSUE TITLE
Deprecate twisted.python.runtime.Platform.isWinNT

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5925]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5925|
|Type|task|
|Created|2012-08-28 21:42:43Z|
|Branch|https://github.com/twisted/twisted/tree/deprecate-iswinnt-5925|

Twisted only supports Windows NT-derived platforms at this point (Windows XP, Windows 7).  See <http://en.wikipedia.org/wiki/Windows_NT>.

This function can only ever return true in a supported environment.

<details><summary>Searchable metadata</summary>

```
trac-id__5925 5925
type__task task
reporter__exarkun exarkun
priority__lowest lowest
milestone__ 
branch__branches_deprecate_iswinnt_5925 branches/deprecate-iswinnt-5925
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__win32 win32
time__1346190163000000 1346190163000000
changetime__1361451075000000 1361451075000000
version__None None
owner__thijs thijs
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

It was only used once, in `python.win32.getProgramsMenuPath` which should be replaced once `isWinNT` is removed.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [36891]) deprecate isWinNT, refs [#5925](https://github.com/twisted/twisted/issues/5925)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_4"></a>|@tomprince set owner to @thijstriemstra|
|-|-|

  1. If this only returns `True` on win32, is there any reason to do anything more complicated than `return True` there? (In which case we can test that fact)
  1. `test_isWinNT` shouldn't emit a deprecation warning.
  1. `python.win32.getProgramsMenuPath` should be fixed to not use `isWinNT` *before* before the deprecation of `isWinNT` (Since it currently lacks tests, this should perhaps be a separate ticket).
  1. Consider using `SynchronousTestCase.callDeprecated` for testing the deprecation.
  1. Consider leaving a comment that `t.p.deprecate` isn't used, to avoid a dependency on that module.

Please resubmit for review after addressing the above issues.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_5"></a>|@tomprince commented|
|-|-|

Also, `assertIn` is preferable to `assertTrue(... in ...)`.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

> Consider using SynchronousTestCase.callDeprecated for testing the deprecation

Development documentation currently recommends `flushWarnings`, not `callDeprecated`.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra set status to `assigned`|
|-|-|

There are also some "nt" references in `twisted/python/dist.py` but that should probably be fixed in separate ticket.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra commented|
|-|-|

(In [37117]) address review comments. refs [#5925](https://github.com/twisted/twisted/issues/5925)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Replying to [tom.prince](#note_4):
>   1. If this only returns `True` on win32, is there any reason to do anything more complicated than `return True` there? (In which case we can test that fact)

Replaced the logic with a simple `isWindows()` check.

>   1. `test_isWinNT` shouldn't emit a deprecation warning.

Fixed.

>   1. `python.win32.getProgramsMenuPath` should be fixed to not use `isWinNT` *before* before the deprecation of `isWinNT` (Since it currently lacks tests, this should perhaps be a separate ticket).

Fixed. Also added tests.

Additionally added a test for `Platform.isKnown` and added some missing docstrings.

Forced a [build](https://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/deprecate-iswinnt-5925).

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_10"></a>|@tomprince set owner to @thijstriemstra|
|-|-|

  1. It looks like there is one relevant twistedchecker error
```
************* Module twisted.python.test.test_win32
W9015: 68,0: Too many blank lines, expected 2
```
  1. There are a bunch of `\r`s in that file.
  1. (optional) I think `twisted.python.win32.getProgramsMenuPath` can probably be allowed to fail on non-windows. i.e. just remove the conditional there. I guess that is an incompatible change though.
  1. `assertNotNull` doesn't exists. The tests probably want to at least assert that the return value is a `C{str}`, since that is what we document.

Please resubmit after addressing the above issues.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_11"></a>|@thijstriemstra commented|
|-|-|

(In [37120]) address review comments. refs [#5925](https://github.com/twisted/twisted/issues/5925)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_12"></a>|@thijstriemstra removed owner|
|-|-|

Replying to [tom.prince](#note_10):
>   1. It looks like there is one relevant twistedchecker error
> ```
> ************* Module twisted.python.test.test_win32
> W9015: 68,0: Too many blank lines, expected 2
> ```

Removed. Tried forcing a new build but the process just hung.. Need to get `twistedchecker` installed locally.

>   1. There are a bunch of `\r`s in that file.

Removed those. Seems these type of changes don't show up in the trac browser.

>   1. (optional) I think `twisted.python.win32.getProgramsMenuPath` can probably be allowed to fail on non-windows. i.e. just remove the conditional there. I guess that is an incompatible change though.

I'd rather handle that in a different ticket (or deprecate these altogether).

>   1. `assertNotNull` doesn't exists. The tests probably want to at least assert that the return value is a `C{str}`, since that is what we document.

Used `assertIsInstance` instead.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_13"></a>|@tomprince set owner to @thijstriemstra|
|-|-|

Looks good. Please merge after the windows builders complete. (There are some unrelated failure on those builders, so check that the tests of the code modified here pass).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_14"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [37185]) Merge deprecate-iswinnt-5925: Platform.isWinNT is deprecated in favor of isWindows.

Author: thijs
Reviewer: tom.prince
Fixes: [#5925](https://github.com/twisted/twisted/issues/5925)

Twisted only supports Windows NT-derived platforms at this point (Windows XP, Windows 7).
